ome_zarr/io.py

```diff
@@ -189,7 +193,7 @@ def get_json(self, subpath: str) -> JSONDict:
             return {}
 
 
-def parse_url(path: str) -> Optional[BaseZarrLocation]:
+def parse_url(path: str, mode: str = "r") -> Optional[BaseZarrLocation]:
```

Comment:
As this function is not used in the write method, maybe I should revert the changes in `io.py`

Reply:
I was working on a simple example for Wei (his use case was "how to turn an existing zarr array into an ome-zarr"). Let's keep this until we have a simple working user example. Maybe `parse_url` can still be the starting point.

<details><summary>previous example</summary>

```
#!/usr/bin/env python
import numpy as np
import zarr as zr

from ome_zarr.io import parse_url
from ome_zarr.reader import OMERO, Reader
from ome_zarr.writer import write_image

import argparse
parser = argparse.ArgumentParser()
parser.add_argument("path")
ns = parser.parse_args()

dtype = np.uint8
mean = 10
shape = (1, 3, 1, 256, 256)
rng = np.random.default_rng(0)
data = rng.poisson(mean, size=shape).astype(dtype)
store = zr.DirectoryStore(ns.path)
grp = zr.group(store)
write_image(data, grp, chunks=(128, 128))
```

</details>